client/src/utils/useAuth.js

```diff
@@ -0,0 +1,28 @@
+import { useState, useEffect } from 'react';
+
+const useAuth = () => {
+  const [logged, setLogged] = useState('loading');
+  const [userId, setUserId] = useState('');
+
+  useEffect(() => {
+    fetch('/api/v1/auth')
+      .then((res) => res.json())
+      .then((result) => {
+        if (result.statusCode === 200) {
+          setLogged(true);
+          setUserId(result.user.id);
+        } else {
+          setLogged(false);
+          setUserId('');
+        }
+      })
+      .catch(() => {
+        setLogged(false);
+        setUserId('');
+      });
+  }, []);
+
+  return [logged, setLogged, userId, setUserId];
```

Comment:
I think you should return the function that calls the API, so we can call it when we click logout
and you also can use an object instead of returning an array, since too many components can be ugly to destructure:
```js
const [ , , userId] = useAuth();
```